tests/test_transaction.py

```diff
@@ -0,0 +1,79 @@
+import unittest
+from happy_bank_core.logic.transaction import Transaction, TransactionException
+from happy_bank_core.logic.account import Account
+
+customer_1 = Account("id123", "John Doe", 100)
+customer_2 = Account("id321", "Johanna Doe", 300)
+
+
+class TestTransaction(unittest.TestCase):
+    def test_insufficient_balance(self):
+        """Tests if function returns false in case of insufficient balance"""
+
+        # Given
+        expected = False
+
+        # When
+        amount = 200
+        result = Transaction.check_balance(customer_1, amount)
+
+        # Then
+        assert result == expected
+
+    def test_sufficient_balance(self):
+        """Tests if function returns true in case of sufficient balance"""
+
+        # Given
+        expected = True
+
+        # When
+        amount = 100
+        result = Transaction.check_balance(customer_1, amount)
+
+        # Then
+        assert result == expected
+
+    def test_zero_withdraw_amount(self):
+        """Tests if function returns false if withdraw amount <= 0"""
+
+        # Given
+        expected = False
+
+        # When
+        amount = 0
+        result = Transaction.check_balance(customer_1, amount)
+
+        # Then
+        assert result == expected
+
+    def test_transfer_throws_exception_if_false(self):
+        """Tests if transfer function throws exception when check_balance func returns false"""
+
+        # When
+        transfer_amount = 0
+
+        # then
+        self.assertRaises(
+            TransactionException, Transaction.transfer, customer_1, customer_2, transfer_amount
+        )
```

Comment:
Could you use clause 'with pytest.raises(...)?

Source: https://docs.pytest.org/en/6.2.x/assert.html#assertions-about-expected-exceptions